src/current-user.ts

```diff
@@ -0,0 +1,10 @@
+import { openmrsFetch } from "./openmrs-fetch";
+
+export function getCurrentUser(errBack) {
```

Comment:
The main purpose of `getCurrentUser` is to prevent many duplicate network requests for the same object. See https://github.com/openmrs/openmrs-rfc-frontend/blob/b2badfb9fd288c8d95a3f5997b33df1beb425ca0/text/0013-api-calls-and-fhir.md#definition.

This file should have `let currentUser` (or similar) defined on the top-level of the file. And the implementation of `getCurrentUser` shouldn't call `openmrsFetch`, wait for the `currentUser` object to be defined and then call the errBack. Additionally, it should recall errBack every time the currentUser object is updated.

Reply:
btw we should also try to decide between errbacks and observables -- the implementation here relies on us choosing. As I expressed over Slack to you, choosing observables is choosing a large learning curve for every developer who ever contributes to OpenMRS. However, there are also some advantages to them too.

Reply:
> And the implementation of getCurrentUser shouldn't call openmrsFetch, but should wait for the currentUser object to be defined and then call the errBack. >

I'm confused by the above. What will be defining currentUser? I thought this function was supposed to do that.